core/src/com/unciv/logic/automation/ConstructionAutomation.kt

```diff
@@ -322,4 +338,48 @@ class ConstructionAutomation(val cityConstructions: CityConstructions){
         }
     }
 
+    private fun addReligousUnit(){
+
+        var modifier = 0f
+
+        val missionary = cityInfo.getRuleset().units.values.asSequence()
+            .firstOrNull { it -> it.canBePurchasedWithStat(cityInfo, Stat.Faith)
+                    && it.getMatchingUniques("Can [] [] times").any { it.params[0] == "Spread Religion"} }
+
+
+        val inquisitor = cityInfo.getRuleset().units.values.asSequence()
+            .firstOrNull { it.canBePurchasedWithStat(cityInfo, Stat.Faith)
+                    && it.hasUnique("Prevents spreading of religion to the city it is next to") }
+
+
+
+        // these 4 if conditions are used to determine if an AI should buy units to spread religion, or spend faith to buy things like new military units or new buildings.
+        // currently this AI can only buy inquisitors and missionaries with faith
+        // this system will have to be reengineered to support buying other stuff with faith
+        if (preferredVictoryType == VictoryType.Domination) return
+        if (civInfo.religionManager.religion?.name == null) return
+        if (preferredVictoryType == VictoryType.Cultural) modifier += 1
+        if (isAtWar) modifier -= 0.5f
+        if (cityInfo.religion.getMajorityReligion()?.name != civInfo.religionManager.religion?.name)
+            return // you don't want to build units of opposing religions.
+
+
+        val citiesNotFollowingOurReligion = civInfo.cities.asSequence()
+            .filterNot { it.religion.getMajorityReligion()?.name == civInfo.religionManager.religion!!.name }
+
+        val buildInqusitor = citiesNotFollowingOurReligion
+            .filter { it.religion.getMajorityReligion()?.name == civInfo.religionManager.religion?.name }
+            .toList().size.toFloat() / 10 + modifier
+
+        val possibleSpreadReligionTargets = civInfo.gameInfo.getCities()
+            .filter { it.getCenterTile().aerialDistanceTo(cityInfo.getCenterTile()) < 30 }
+
+        val buildMissionary = possibleSpreadReligionTargets.toList().size.toFloat() / 15 + modifier
+
+        if (buildMissionary > buildInqusitor && missionary != null && inquisitor != null) faithConstruction.add(missionary)
```

Comment:
Why do you check here if inquisitor is not null?

Reply:
debug code